### PR TITLE
Fix break in a condition

### PR DESCRIPTION
**Pull request by DJMcNab**
The part of fixing #290.

### CODE REVIEW

File: crates/ra_syntax/src/grammar/expressions/atom.rs [hunk 1]
```diff
@@ -427,12 +430,20 @@ fn continue_expr(p: &mut Parser) -> CompletedMarker {
...
+    // test break_ambiguity
+    // fn foo(){
+    //     if break {}
+    //     while break {}
+    //     for i in break {}
+    //     match break {}
+    // }
+    if r.forbid_structs && p.at_ts(EXPR_FIRST_NO_BLOCK) || !r.forbid_structs && p.at_ts(EXPR_FIRST)
+    {
         expr(p);
```
Comment (matklad) [inline crates/ra_syntax/src/grammar/expressions/atom.rs]: Should we pass `r` down to `expr` here?

Is `if break S {} {}` parsed correctly?

Comment (DJMcNab) [reply crates/ra_syntax/src/grammar/expressions/atom.rs]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=55e91f597b049520bb4b7047dbe136f6 seems to suggest it's allowed, and we do allow it currently. I think we're alright. I might rename the field on Restrictions though to be more descriptive.

File: crates/ra_syntax/src/grammar/expressions.rs [hunk 2]
```diff
@@ -5,6 +5,7 @@ pub(super) use self::atom::{literal, LITERAL_FIRST};
 use super::*;
 
 const EXPR_FIRST: TokenSet = LHS_FIRST;
+const EXPR_FIRST_NO_BLOCK: TokenSet = LHS_FIRST_NO_BLOCK;
```
Comment (matklad) [inline crates/ra_syntax/src/grammar/expressions.rs]: So, this is basically `EXPR_FIRST - {`, right? Let's perhaps just check `EXPR_FIRST.contains(p.current()) && p.current() != L_CURLY`?

Comment (DJMcNab) [reply crates/ra_syntax/src/grammar/expressions.rs]: That's a better way of doing it :man_facepalming:.

**Pull request merged**